### ISSUE TITLE
ComboBox popup close bug when setEmptySelectionAllowed(false)

### ISSUE THREAD

**Issue by sirbris** [vaadin/framework#11088]
Vaadin 8.5.0 and prior

Preconditions:
ComboBox with setTextInputAllowed(true), setEmptySelectionAllowed(false) and no selection yet.

Open CombBox and hover over items but do NOT select any of them. Click outside the combobox so the popup closes. When you now try to open the ComboBox again it closes immediately afterwards.

Try this here:
https://demo.vaadin.com/sampler/#ui/data-input/multiple-value/combo-box

**Comment by steeeffen** [comment]
annoying bug, can reproduce it

**Comment by anasmi** [comment]
Hi,

I have tried to reproduce issue in Sampler, and then also using 

```
 ComboBox<String> cb=new ComboBox<>();
        cb.setItems("Test1","Test3","Test2","test2");
        cb.setTextInputAllowed(true);
        cb.setEmptySelectionAllowed(false);
        layout.addComponent(cb);
```

Through, the popup is not closed.

What OS and browsers you are using? (Tested on windows + IE/Chrome)

**Comment by OlliTietavainenVaadin** [comment]
Tested with 8.5.1 sampler with latest Chrome, IE11, Edge and Firefox, could not reproduce.

**Comment by sirbris** [comment]
Could also not reproduce with current Vaadin version... 8.5.1 seems to fix that issue!

**Comment by iLutor** [comment]
Did someone try to test with the property setNewItemProvider() and NewItemProvider ?, I allways get this error and my version is 8.4.3

ComboBox cbx_AllPositions = new ComboBox<ContactProviderPositionDTO>();
		cbx_AllPositions.setItemCaptionGenerator(ContactProviderPositionDTO::getName);
		cbx_AllPositions.setTextInputAllowed(true);
		cbx_AllPositions.setEmptySelectionAllowed(false);
		cbx_AllPositions.setNewItemProvider((NewItemProvider<ContactProviderPositionDTO> & Serializable) e -> this.newItemHandler(e));
		cbx_AllPositions.setItems(allPositionsList);

**Comment by tsuoanttila** [comment]
@iLutor There was a fix for this issue in 8.5.1, please update to the latest version to get this fix.
